webapp/src/components/Profile.tsx

```diff
@@ -25,20 +25,73 @@ import CompetencyRatings from './CompetencyRatings';
  * User data is currently hardcoded but it will be pulling data from the database (github_users)
  */
 
-const user = {
-  name: 'Matthew Morenez',
-  email: 'profile@example.com',
-  avatar:
-    'https://media.volinspire.com/images/95/e4/99/95e499b759ba57975a61c7bf66a3414dd5a2625e_profile.jpg',
-  github: 'https://github.com',
-  website: 'https://example.com',
-  summary:
-    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officia, sit impedit? Cupiditate veniam eaque suscipit eligendi. Sint delectus enim earum non repellendus nihil numquam libero odit temporibus et, natus eaque?',
-  linkedIn: 'https://linkedin.com',
-};
+//
+import { useState } from 'react';
+import useApiData from '../helpers/useApiData';
+import { EntityId, Profile as APIProfile } from '../types/api';
+import SessionContext from './SessionContext';
+
+// const user = {
+//   name: 'Matthew Morenez',
+//   email: 'profile@example.com',
+//   avatar:
+//     'https://media.volinspire.com/images/95/e4/99/95e499b759ba57975a61c7bf66a3414dd5a2625e_profile.jpg',
+//   github: 'https://github.com',
+//   website: 'https://example.com',
+//   summary:
+//     'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Officia, sit impedit? Cupiditate veniam eaque suscipit eligendi. Sint delectus enim earum non repellendus nihil numquam libero odit temporibus et, natus eaque?',
+//   linkedIn: 'https://linkedin.com',
+// };
 
 const Profile = () => {
-  const greeting = getGreeting(user.name);
+  const { principalId } = useContext(SessionContext);
+  console.log({ id: principalId });
+  const [changedUserDataIds, setChangedUserDataIds] = useState<EntityId[]>([]);
+  const { data: userData, error } = useApiData<APIProfile[]>({
+    deps: [changedUserDataIds],
+    path: `/profile/${principalId}`,
+    sendCredentials: true,
+  });
+  console.log('id', userData && userData[0].github_accounts[0].avatar_url);
+  {
+    const ref = [
+      {
+        id: 1,
+        full_name: null,
+        email_address: null,
+        bio: null,
+        github_accounts: [
+          {
+            github_id: 26878542,
+            username: 'SiriusLL',
+            full_name: null,
+            bio: 'Full Stack Junior Web Developer ~ “The Way is not in the sky; the Way is in the heart.” --Buddha',
```

Comment:
Why are there two full_name and bio attributes? So that they can be changed in the principals table later and keep the GitHub data as it originally was?

Reply:
Correct! So for instance, GitHub tells the webapp that my `full_name` is "Spencer W.", and I would want to override that in Rhizone. But next time I log in with my GitHub (if the cookie expired, for instance), the value we get back from GitHub may override my override. So it's easier to have a `full_name` variable in both `github_users` as well as in `principals`.